### PR TITLE
Targeted documentation updates in autograd.functional

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72111

For vectorize flag:
- Advertises the use of functorch

For autograd.functional.jvp:
- Advertises the use of functorch and the low-level jvp API, both of
which will be more performant than the double backprop trick.

Test Plan:
- view docs

Differential Revision: [D33918065](https://our.internmc.facebook.com/intern/diff/D33918065)

### CODE REVIEW

**Review by albanD — APPROVED** [review]
SGTM

**Comment by zou3519** [comment]
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33918065).

**Review by soulitzer — APPROVED** [review]

**Comment by zou3519** [comment]
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33918065).

File: torch/autograd/functional.py [hunk 1]
```diff
@@ -323,6 +323,13 @@ def jvp(func, inputs, v=None, create_graph=False, strict=False):
             jvp (tuple of Tensors or Tensor): result of the dot product with
             the same shape as the output.
 
+    Note:
+        ``autograd.functional.jvp`` computes the jvp by using the backward of
+        the backward (sometimes called the double backwards trick). This is not
+        the most performant way of computing the jvp. Please consider using
+        `functorch's jvp <https://github.com/pytorch/functorch#jvp>`_
+        or the low-level forward-mode AD API instead.
```
Comment (albanD) [inline torch/autograd/functional.py:331]: You can link to https://pytorch.org/docs/master/autograd.html#forward-mode-automatic-differentiation here

**Comment by zou3519** [comment]
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33918065).

File: torch/autograd/functional.py (hunk 1 above)
Comment (zou3519) [reply torch/autograd/functional.py]: Yup, it took me a while to figure out how to actually get that cross-reference to work.

**Comment by zou3519** [comment]
I ended up building the documentation locally because I think there is something wrong with the docs build job in CI. It looks fine:
image

**Pull request closed without merging**